src/coreclr/jit/objectalloc.cpp

```diff
@@ -597,6 +618,38 @@ void ObjectAllocator::MarkEscapingVarsAndBuildConnGraph()
 
             return Compiler::fgWalkResult::WALK_CONTINUE;
         }
+
+        Compiler::fgWalkResult PostOrderVisit(GenTree** use, GenTree* user)
+        {
+            GenTree* const tree = *use;
+            if (tree->OperIsLocalStore())
```

Comment:
Why is this part done in post order?

Reply:
Same answer as above: https://github.com/dotnet/runtime/pull/113808#discussion_r2010754388

We need to know that all the descendant lcl var nodes have already tried to trace their paths up to this node.

Reply:
Ah ok, `CanLclVarEscapeViaParentStack` is called from this visitor in the preorder... Now things make more sense to me.

Reply:
I suppose I could move the current preorder processing to postorder too... will leave as-is for now.

Reply:
Yeah... Or even more radically this visitor could work similarly to local morph, tracking abstract information about operands on its own stack and escaping them from the parent instead of the child.